detection-rules/suspicious_sender_display_name_procedurally_generated_blob.yml

```
@@ -0,0 +1,35 @@
+name: "Suspicious sender display name with long procedurally generated text blob"
+description: "This rule identifies sender display names containing long strings of nonsensical or procedurally generated characters, which are often used in phishing or spam campaigns for campaign tracking and identification, as well as to bypass detection filters."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and regex.icontains(sender.display_name, '\b[\w\p{L}\p{N}]{35,}\b')
+  and not regex.icontains(sender.display_name, '_bot_[a-f0-9]{32}\)')
+  and not regex.match(sender.display_name, '\b[\w\p{L}\p{N}]{35,}\b')
+  
```

avoid matching on unparsed sender values
```suggestion
  and not (sender.email.email == "" or sender.email.domain.valid == false)
```